src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -312,8 +312,184 @@ void bulk_load_service::partition_bulk_load(const std::string &app_name, const g
 void bulk_load_service::on_partition_bulk_load_reply(error_code err,
                                                      const bulk_load_request &request,
                                                      const bulk_load_response &response)
+{
+    const std::string &app_name = request.app_name;
+    const gpid &pid = request.pid;
+    const rpc_address &primary_addr = request.primary_addr;
+    int32_t interval = bulk_load_constant::BULK_LOAD_REQUEST_INTERVAL;
+
+    if (err != ERR_OK) {
+        derror_f("app({}), partition({}) failed to recevie bulk load response, error = {}",
+                 pid.get_app_id(),
+                 pid,
+                 err.to_string());
+        try_rollback_to_downloading(app_name, pid);
+        try_resend_bulk_load_request(app_name, pid, interval);
```

Comment:
How to deal with a nonrecoverable error, e.g. files in remote storage is conrrupt?

Reply:
On L353 ~ L363, the app and partitions' bulk load status will be set as bulk_load_failed (will be implemented in function`handle_bulk_load_failed`), and resend request to replica server to cleanup bulk load states and context.